modules/collect/src/main/java/com/opengamma/strata/collect/result/ValueWithFailures.java

```diff
@@ -158,6 +158,62 @@
     return Collectors.reducing(ValueWithFailures.of(identityValue), combiningValues(operator));
   }
 
+  /**
+   * Returns a collectors that can be used to create a ValueWithFailure instance with a list of success values
+   * from a stream of ValueWithFailures instances.
+   *
+   * @param <T>  they type of the success value in the {@link ValueWithFailures}
+   * @return a {@link Collector}
+   */
+  public static <T> Collector<ValueWithFailures<T>, ?, ValueWithFailures<ImmutableList<T>>> toValueListWithFailures() {
+    return Collector.of(
+        StreamBuilder<T>::new,
+        StreamBuilder::addSingle,
+        StreamBuilder::combining,
+        StreamBuilder::build);
+  }
+
+  // Utility class to help combine ValueWithFailures
+  private static final class StreamBuilder<T> {
+
+    private final ImmutableList.Builder<T> values = ImmutableList.builder();
+    private final ImmutableList.Builder<FailureItem> failures = ImmutableList.builder();
+
+    private void addSingle(ValueWithFailures<T> item) {
+      values.add(item.getValue());
+      failures.addAll(item.getFailures());
+    }
+
+    private StreamBuilder<T> combining(StreamBuilder<T> builder) {
+      values.addAll(builder.values.build());
+      failures.addAll(builder.failures.build());
+      return this;
+    }
+
+    private ValueWithFailures<ImmutableList<T>> build() {
+      return ValueWithFailures.of(values.build(), failures.build());
+    }
+  }
+
+  /**
+   * Converts a list of value with failures to a single value with failures, combining the values into a list.
+   * <p>
+   * Effectively, this converts {@code List<ValueWithFailures<T>>} to {@code ValueWithFailures<List<T>>}.
+   *
+   * @param <T>  they type of the success value in the {@link ValueWithFailures}
+   * @return a new instance with a list of success values
+   */
+  public static <T> ValueWithFailures<ImmutableList<T>> combineAsList(List<? extends ValueWithFailures<? extends T>> items) {
```

Comment:
Looking at the equivalent method in `Guavate`, it returns `List`, not `ImmutableList`.

Reply:
Maybe named `combinedValuesAsList()` to match Guavate